src/lib.rs

```diff
@@ -254,96 +253,37 @@ fn run(args: Vec<String>, write_mode: WriteMode) {
     rustc_driver::run_compiler(&args, &mut call_ctxt);
 }
 
-#[cfg(not(test))]
-fn main() {
-    let args: Vec<_> = std::env::args().collect();
-    //run(args, WriteMode::Display);
+pub fn format(args: Vec<String>) {
     run(args, WriteMode::Overwrite);
-    std::env::set_exit_status(0);
-
-    // TODO unit tests
-    // let fmt = ListFormatting {
-    //     tactic: ListTactic::Horizontal,
-    //     separator: ",",
-    //     trailing_separator: SeparatorTactic::Vertical,
-    //     indent: 2,
-    //     h_width: 80,
-    //     v_width: 100,
-    // };
-    // let inputs = vec![(format!("foo"), String::new()),
-    //                   (format!("foo"), String::new()),
-    //                   (format!("foo"), String::new()),
-    //                   (format!("foo"), String::new()),
-    //                   (format!("foo"), String::new()),
-    //                   (format!("foo"), String::new()),
-    //                   (format!("foo"), String::new()),
-    //                   (format!("foo"), String::new())];
-    // let s = write_list(&inputs, &fmt);
-    // println!("  {}", s);
 }
 
-
-#[cfg(test)]
-mod test {
-    use std::collections::HashMap;
+pub fn idempotent_check(args: Vec<String>) -> Result<(), HashMap<String, String>> {
```

Comment:
I think this whole function should be in tests/idem too, I don't think it needs to be here.